scripts/airgap.sh

```diff
@@ -0,0 +1,274 @@
+  --insecure      For pull command skip kwctl certificate verification (an alternative to sources.yaml file)
+                  For install disable policyServer verification of registry
+
+Commands:
+  list     Generate json list of dependencies. You can redirect it to a file
+  pull     Pull components from list to current directory
+  push     Push components in list from current directory to local registry
+  install  Install helm charts configured with local registry
+
+Requirements: (https://docs.kubewarden.io/operator-manual/airgap/requirements)
+  - jq, kwctl, helm, docker
+  - kubewarden helm repository
+
+Examples:
+  Generate list of dependencies required by airgap. Save in file for other commands:
+    $PROG list --cert-manager | tee file.json
+
+  Pull images and policies and save them to archives in current directory:
+    $PROG pull --list file.json
+
+  Push images and policies from created archives to local registry. Run in DRY mode:
+    $PROG push --list file.json --registry=localhost:5123 --dry
+
+  Helm install charts from local files. Registry is used for recommendedPolicies setup:
+    $PROG install --list file.json --registry 172.18.0.4:5000"
+
+# ==================================================================================================
+# Helper Functions & traps
+
+shopt -s expand_aliases
+alias curl="curl -# -L"
+
+# Print error message (or usage) and exit
+function error { echo -e "${1:-$USAGE}" >&2; exit 1; }
+
+# Print green colored message
+function step { printf -- "\e[32m# %s\e[0m\n" "${*}"; }
+
+# shellcheck disable=SC2015
+# Print command that would be executed in DRY mode
+function dry { [ -v DRY ] && echo "- $*" || "$@"; }
+
+# Transform lines to json array key -> [lines]
+function to_json { cat < /dev/stdin | jq -s -c -R --arg key "$1" '{ ($key): split("\n") | map(select(length > 0)) }'; }
+
+# Get archive location from chart line: https://charts.kubewarden.io/kubewarden-crds:1.2.3 -> kubewarden-crds-1.2.3.tgz
+# function chart_file { jq -er --arg chart "$1" '$chart +"-"+ (.charts[] | select(contains($chart)) | split(":")[-1]) +".tgz"' <<< "$LIST"; }
+function chart_file { printf -- '%s\n' "${charts[@]##*/}" | grep "$1" | tr ':' '-' | sed 's/$/.tgz/'; }
+
+# Print line with error
+trap 'echo "Error on ${BASH_SOURCE}:${LINENO} $(sed -n "${LINENO} s/^\s*//p" ${BASH_SOURCE})"' ERR
+
+# ==================================================================================================
+# Generate json from all components required by airgap
+
+function command_list {
+   local tag url urls images policies charts
+
+   tag=$(helm search repo kubewarden/kubewarden-defaults -o json | jq -er 'first.version')
+   urls=$(curl "https://api.github.com/repos/kubewarden/helm-charts/releases/tags/kubewarden-defaults-$tag" | jq -er '.assets[].browser_download_url')
+
+   # images
+   url=$(grep 'images.txt$' <<< "$urls" | tail -1)
+   images=$(curl "$url" | to_json 'images')
+   if [ -v CERTMAN ]; then
```

Comment:
I see where you are going by taking care of cert-manager in list and install subcommands, but I think it's better to leave cert-manager out of this script and have it directly in our CI tests, it would be super helpful there.

The current code is taking the images the latest cert-manager helm chart, and installing the latest helm chart, but that may or may not be desirable. Also the users may need to configure cert-manager in their own way for example. And by configuring a helm chart, you may change the usage of images, too. I think its ok to opt-out to all of this, and refer users to upstream cert-manager for its airgapped deployment.